compiler/test/test_end_to_end.re

```diff
@@ -21,12 +21,41 @@ let t = (~todo=?, name, program, expected) =>
   name >:: wrap_todo(todo) @@ test_run(program, name, expected);
 let tc = (~todo=?, name, program, expected) =>
   name >:: wrap_todo(todo) @@ test_run(~cmp=exists, program, name, expected);
-let tgc = (~todo=?, name, heap_size, program, expected) =>
-  name >:: wrap_todo(todo) @@ test_run(~heap_size, program, name, expected);
+
+let make_gc_program = (program, heap_size) => {
+  {|
+    import WasmI32 from "runtime/unsafe/wasmi32"
+    import Malloc from "runtime/malloc"
+    import Memory from "runtime/unsafe/memory"
+    @disableGC
+    let leak = () => {
+      // find current memory pointer, subtract space for two malloc headers + 1 GC header
+      let offset = WasmI32.sub(Memory.malloc(8n), 24n)
+      // Calculate how much memory is left
+      let availableMemory = WasmI32.sub(offset, Malloc._RESERVED_RUNTIME_SPACE)
+      // Calculate how much memory to leak
+      let toLeak = WasmI32.sub(availableMemory, |}
+  ++ string_of_int(heap_size)
```

Comment:
Is it too nitpicky to want this to be a Sprintf on against the multiline string?

Reply:
Nope, not at all. For some reason, I thought this was going to be more readable than a `sprintf` but that doesn't make any sense.